File: bot.js
require('dotenv').config();
const { Telegraf } = require('telegraf');
const axios = require('axios');

const bot = new Telegraf(process.env.BOT_TOKEN);

// Sample products
const products = [
  { id: 1, name: 'Product A', price: 1000 }, // Price in cents (USD 10.00)
  { id: 2, name: 'Product B', price: 2000 }, // Price in cents (USD 20.00)
  { id: 3, name: 'Product C', price: 3000 }  // Price in cents (USD 30.00)
];

// Start command
bot.start((ctx) => {
  ctx.reply('Welcome to our store! Use /shop to view products.');
});

// Show available products
bot.command('shop', (ctx) => {
  let message = 'Available Products:\n\n';
  products.forEach((product) => {
    message += `${product.name} - $${(product.price / 100).toFixed(2)}\n`;
    message += `/buy_${product.id} - Buy Now\n\n`;
  });
  ctx.reply(message);
});

// Handle purchase commands
products.forEach((product) => {
  bot.command(`buy_${product.id}`, async (ctx) => {
    try {
      const chatId = ctx.chat.id;
      const priceInUSD = product.price / 100;

      const paymentLink = await createCryptoPaymentLink(product.name, priceInUSD, chatId);
      
      ctx.replyWithMarkdown(
        `You are purchasing *${product.name}* for $${priceInUSD}. [Pay Now](${paymentLink})`
      );
    } catch (error) {
      console.error(error);
      ctx.reply('Error processing your order. Please try again later.');
    }
  });
});

// Function to create CryptoBot payment link
async function createCryptoPaymentLink(productName, priceInUSD, userId) {
  const CRYPTOPAY_API_KEY = process.env.CRYPTOPAY_API_KEY;
  const CRYPTOPAY_BOT_USERNAME = process.env.CRYPTOPAY_BOT_USERNAME;
  
  const amount = priceInUSD; // Payment amount in USD
  const payload = `user_${userId}_product_${productName}`; // Payload to track the payment

  const response = await axios.post(
    `https://pay.crypt.bot/api/createInvoice`,
    {
      asset: 'USDT',  // You can use other crypto assets supported by CryptoBot
      amount: amount,
      description: `Payment for ${productName}`,
      payload: payload,
      paid_btn_name: "viewItem",
      paid_btn_url: "https://yourstore.com/thankyou"
    },
    {
      headers: {
        'Crypto-Pay-API-Token': CRYPTOPAY_API_KEY
      }
    }
  );

  return `https://t.me/${CRYPTOPAY_BOT_USERNAME}?start=${response.data.result.invoice_id}`;
}

// Start the bot
bot.launch();

console.log('Bot is running...');

File: bot.py
import telebot
import requests

# Telegram bot token
BOT_TOKEN = 'YOUR_TELEGRAM_BOT_TOKEN'
bot = telebot.TeleBot(BOT_TOKEN)

# CryptoBot API token
CRYPTOPAY_API_KEY = 'YOUR_CRYPTOPAY_API_KEY'
CRYPTOPAY_API_URL = 'https://pay.crypt.bot/api/'

# Sample products (name: price in USDT)
products = {
    "Product 1": {"price": 5, "description": "Description of product 1"},
    "Product 2": {"price": 10, "description": "Description of product 2"},
    "Product 3": {"price": 20, "description": "Description of product 3"}
}

# Start command
@bot.message_handler(commands=['start'])
def send_welcome(message):
    bot.send_message(
        message.chat.id,
        "Welcome to the shop! Type /products to see our catalog."
    )

# List products
@bot.message_handler(commands=['products'])
def list_products(message):
    product_list = "Available products:\n\n"
    for product, details in products.items():
        product_list += f"{product}: {details['price']} USDT\n"
    product_list += "\nReply with /buy <Product Name> to purchase."
    bot.send_message(message.chat.id, product_list)

# Handle buying a product
@bot.message_handler(commands=['buy'])
def buy_product(message):
    try:
        product_name = message.text.split("/buy ", 1)[1]
        if product_name not in products:
            bot.send_message(message.chat.id, "Product not found. Please try again.")
            return

        product = products[product_name]
        create_payment(message.chat.id, product_name, product['price'])

    except IndexError:
        bot.send_message(message.chat.id, "Please specify a product name. Example: /buy Product 1")

# Create a payment link using CryptoBot API
def create_payment(chat_id, product_name, amount):
    payload = {
        "token": CRYPTOPAY_API_KEY,
        "currency": "USDT",
        "amount": amount,
        "description": f"Payment for {product_name}",
        "paid_btn_name": "viewItem",
        "paid_btn_url": "https://example.com",  # URL after payment, can be a receipt page
        "payload": product_name
    }

    response = requests.post(f"{CRYPTOPAY_API_URL}createInvoice", json=payload)

    if response.status_code == 200:
        data = response.json()
        payment_url = data['result']['pay_url']
        bot.send_message(chat_id, f"Click the link to pay for {product_name}: {payment_url}")
    else:
        bot.send_message(chat_id, "Failed to create payment. Please try again later.")

# Run the bot
bot.polling()
